packages/core/src/delegation/DelegationRootNode.ts

```diff
@@ -19,11 +19,25 @@ import DelegationBaseNode from './Delegation'
 import DelegationNode from './DelegationNode'
 import { getChildren } from './DelegationNode.chain'
 import { query, revoke, store } from './DelegationRootNode.chain'
+import * as DelegationRootNodeUtils from './DelegationRootNode.utils'
 
 const log = ConfigService.LoggingFactory.getLogger('DelegationRootNode')
 
 export default class DelegationRootNode extends DelegationBaseNode
   implements IDelegationRootNode {
+  /**
+   * Instantiates a new DelegationRootNode from the given [[IDelegationRootNode]].
+   *
+   * @param delegationRootNodeInput IDelegationBaseNode to instantiate the new delegation node from.
+   *
+   * @returns An instantiated DelegationRootNode.
+   */
+  public static async fromDelegationRootNode(
+    delegationRootNodeInput: IDelegationRootNode
+  ): Promise<DelegationRootNode> {
+    return new DelegationRootNode(delegationRootNodeInput)
+  }
```

Comment:
why is this async - and what do we need it for?

Reply:
Originally, I was going to have the constructor check if there is an `id` and if not generate one.

The `fromDelegationRootNode` was for pre-constructed delegations would take the id as a para, which would have checked the if the `id` was valid etc. 

With the purposed changes to construct the `id` in the mode, I removed it. 

I would say that this can be removed.

Reply:
I also removed the `fromDelegationNode` method from the `DelegationNode` class.